build error with llvm: samples/subsys/fs/littlefs

**Describe the bug**
When building with llvm:

```
ccache /usr/lib64/ccache/clang -DBUILD_VERSION=v2.5.0-rc3-3-ge620568d6dc0 -DKERNEL -D_POSIX_C_SOURCE=200809 -D_XOPEN_SOURCE=600 -D_XOPEN_SOURCE_EXTENDED -D__ZEPHYR__=1 -I/home/nashif/Work/zephyrproject/zephyr/subsys/fs -I/home/nashif/Work/zephyrproject/zephyr/include -Izephyr/include/generated -I/home/nashif/Work/zephyrproject/zephyr/soc/posix/inf_clock -I/home/nashif/Work/zephyrproject/zephyr/boards/posix/native_posix -I/home/nashif/Work/zephyrproject/modules/fs/littlefs -Og -imacros /dev/shm/twister-out/native_posix/samples/subsys/fs/littlefs/sample.filesystem.littlefs/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -imacros /home/nashif/Work/zephyrproject/zephyr/include/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-typedef-redefinition -Wno-pointer-sign -Wpointer-arith -Wno-sometimes-uninitialized -Wno-shift-overflow -Wno-missing-braces -Wno-self-assign -Wno-address-of-packed-member -Wno-unused-function -Wno-initializer-overrides -Wno-section -Wno-unknown-warning-option -Wno-unused-variable -Wno-format-invalid-specifier -Wno-gnu -Wno-tautological-compare -Werror=implicit-int -Werror -fno-asynchronous-unwind-tables -fno-pie -fno-pic -fno-strict-overflow -fmacro-prefix-map=/home/nashif/Work/zephyrproject/zephyr/samples/subsys/fs/littlefs=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/nashif/Work/zephyrproject/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/nashif/Work/zephyrproject=WEST_TOPDIR -ffunction-sections -fdata-sections -m32 -include /home/nashif/Work/zephyrproject/zephyr/arch/posix/include/posix_cheats.h -std=c11 -MD -MT zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/littlefs_fs.c.obj -MF zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/littlefs_fs.c.obj.d -o zephyr/subsys/fs/CMakeFiles/subsys__fs.dir/littlefs_fs.c.obj -c /home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:47:5: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if FC_ON_HEAP
    ^
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:44:20: note: expanded from macro 'FC_ON_HEAP'
#define FC_ON_HEAP defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)      \
                   ^
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:88:5: error: macro expansion producing 'defined' has undefined behavior [-Werror,-Wexpansion-to-defined]
#if FC_ON_HEAP
    ^
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:44:20: note: expanded from macro 'FC_ON_HEAP'
#define FC_ON_HEAP defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)      \
                   ^
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:104:5: error: macro expansion producing 'defined' has undefine
d behavior [-Werror,-Wexpansion-to-defined]
#if FC_ON_HEAP
    ^
/home/nashif/Work/zephyrproject/zephyr/subsys/fs/littlefs_fs.c:44:20: note: expanded from macro 'FC_ON_HEAP'
#define FC_ON_HEAP defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)      \
                   ^
3 errors generated.
```

**To Reproduce**
Steps to reproduce the behavior:
1. export ZEPHYR_TOOLCHAIN_VARIANT=llvm
2. twister -T samples/subsys/fs/littlefs -p native_posix


**Expected behavior**
build is successful

*
**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain  LLVM 11
 - Commit SHA or Version used

--- a/subsys/fs/littlefs_fs.c
+++ b/subsys/fs/littlefs_fs.c
@@ -41,8 +41,12 @@ static K_MEM_SLAB_DEFINE(lfs_dir_pool, sizeof(struct lfs_dir),
  * mem_pool or heap Kconfig options then we need to use a heap.
  * Otherwise we can use a fixed region.
  */
-#define FC_ON_HEAP defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)	\
+#if defined(CONFIG_FS_LITTLEFS_FC_MEM_POOL)		\
 	|| ((CONFIG_FS_LITTLEFS_FC_HEAP_SIZE - 0) > 0)
+#define FC_ON_HEAP 1
+#else
+#define FC_ON_HEAP 0
+#endif
 
 #if FC_ON_HEAP
 

